### DISCUSSION TITLE
Different Download Options for Original and Common Formats

### DISCUSSION THREAD

**Discussion by Johnny-31** [Radiokot/photoprism-android-client#166] [Ideas]
Hi,

Thank you for the amazing work on the Gallery app for PhotoPrism! I’ve noticed a recent change in the app’s behavior regarding image downloads. The current setup automatically downloads the original file when I use the download button, which in my case is often a RAW image. While this is great for preserving quality, it creates challenges when trying to share images directly through social media or other apps, as many do not support RAW formats.

I’d like to suggest a couple of improvements to address this:

1. Separate Behavior for Share and Download Buttons:

The "Download" button could continue to download the original file.

The "Share" button could automatically select a more common format (e.g., JPEG) before sharing.

2. Two Download Buttons or a Setting Toggle:

Add two download buttons: one for downloading the original file and another for downloading a converted version (e.g., JPEG).

Alternatively, provide an option in the settings to toggle between downloading the original or a common format.

These changes would make the app more versatile and user-friendly, especially for those of us who often need images in widely supported formats.

I’d love to hear your thoughts on this suggestion. Thanks again for your hard work!

**Comment by Radiokot** [comment]
Hi, this is a fair point. I tried sharing RAWs from the public demo to Telegram and Instagram, it worked, so I thought there was no problem. I probably would go with your first suggestion. Meanwhile, if it is critical for you, you can downgrade by installing the previous version from GitHub: https://github.com/Radiokot/photoprism-android-client/releases/tag/1.32.0(48)

**Reply by Johnny-31** [reply]
I appreciate you looking into this and testing the functionality. You’re right that the issue depends on the RAW format. While some platforms may support certain RAW formats, many do not. For example, I shoot in JPEG and Canon’s CR3 format (CRAW 3). Unfortunately, apps like Instagram and WhatsApp cannot work with CR3 files.

Thank you again for your work on this app—I’m looking forward to future updates!

**Comment by Radiokot** [comment]
Hey, please check out this 1.33.0 clone with the ability to change the sharing behavior for RAWs
https://send.bitwarden.com/#8dqg_-CQ6EKdr7JVAQzl_w/M49xzkm5kB7Jzk_4ABNTgw

**Reply by Johnny-31** [reply]
Hi, i'm sorry für my late Response.
Unfortunately I cannot install apps form other resources than the PlayStore so it is impossible for me to try the new version. Maybe I can try it with an Android emulator on windows if it is important for you! Otherwise I am thrilled to see the new feature in one of the next releases!

**Reply by Radiokot** [reply]
It is already in Google Play, version 1.34.0

**Reply by Johnny-31** [reply]
Sorry, I have been been offline nearly the whole last weeks so my app has not been updated till now!
I reviewed the solution and I think it is the best solution possible. The switch in the settings is pretty good too!
By opening a picture through Lightroom it downloads the raw file - perfect! 
Pretty good update I think!
